app/Events/PostSaving.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Events;
+
+use App\Models\Post;
+use Illuminate\Broadcasting\InteractsWithSockets;
+use Illuminate\Broadcasting\PrivateChannel;
+use Illuminate\Foundation\Events\Dispatchable;
+use Illuminate\Queue\SerializesModels;
+
+class PostSaving
+{
+    use Dispatchable, InteractsWithSockets, SerializesModels;
+
+    public $post;
```

Comment:
Tenta sempre tipar os atributos pra ti ter type check e typehint do teu editor
```suggestion
    public Post $post;
```